phpseclib/Common/Functions/Strings.php

```diff
@@ -33,10 +33,27 @@ class Strings
      * @access public
      * @return string
      */
-    static function shift(&$string, $index = 1)
+    public static function shift(&$string, $index = 1)
     {
         $substr = substr($string, 0, $index);
         $string = substr($string, $index);
         return $substr;
     }
+
+    /**
+     * String Pop
+     *
+     * Inspired by array_pop
+     *
+     * @param string $string
+     * @param int $index
+     * @access public
+     * @return string
+     */
+    public static function pop(&$string, $index = 1)
```

Comment:
It's called. eg.

https://github.com/terrafrost/phpseclib/blob/a25c8411087b1bf3531c0aab919470252b41147a/phpseclib/Crypt/Common/SymmetricKey.php#L878